server/src/api/site.rs

```diff
@@ -300,9 +306,35 @@ impl Perform for Oper<EditSite> {
 
     let found_site = blocking(pool, move |conn| Site::read(conn, 1)).await??;
 
+    let icon = match &data.icon {
+      // An empty string is an erase
+      Some(icon) => {
+        if !icon.eq("") {
+          Some(Some(icon.to_owned()))
+        } else {
+          Some(None)
+        }
+      }
+      None => Some(found_site.icon),
+    };
```

Comment:
You can easily extract this into a helper function.

Reply:
Ok done now.